### PR TITLE
Add feature flag for gating NPS Survey

### PR DESCRIPTION
**Pull request by adhiamboperes**
## Explanation
Add feature flag for gating NPS Survey

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt [hunk 1]
```diff
@@ -93,7 +96,9 @@ class ApplicationLifecycleObserver @Inject constructor(
   /** Occurs when application comes to foreground. */
   @OnLifecycleEvent(Lifecycle.Event.ON_START)
   fun onAppInForeground() {
-    applicationLifecycleListeners.forEach(ApplicationLifecycleListener::onAppInForeground)
+    if (enableNpsSurvey.value) {
+      applicationLifecycleListeners.forEach(ApplicationLifecycleListener::onAppInForeground)
```
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt]: For this & the next block, couldn't non-NPS listeners be theoretically bound to the set? I wonder if we should instead use the flag to gate the logic of the listener itself.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @adhiamboperes! Just had one thought--PTAL.

File: domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt (hunk 1 above)
Comment (adhiamboperes) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt]: Does this mean only calling the timer functions in the listener impl if the feature flag is enabled? Please see my latest commit.

Comment (BenHenning) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt]: If that's the only listener that's required to trigger the functionality, then yes I think so.

**Review by BenHenning — APPROVED** [review]
Thanks @adhiamboperes! This LGTM.

**Pull request merged**